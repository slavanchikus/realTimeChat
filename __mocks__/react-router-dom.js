import React from 'react';

const rrd = require('react-router-dom');
rrd.HashRouter = ({ children }) => <div>{children}</div>;

module.exports = rrd;
