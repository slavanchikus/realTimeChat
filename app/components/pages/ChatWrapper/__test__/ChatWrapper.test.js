import React from 'react';
import { mount, configure } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

import ChatWrapper from '../ChatWrapper';

configure({ adapter: new Adapter() });

describe('ChatWrapper', () => {
  let props;
  const chatWrapper = newProps => (
    <ChatWrapper {...newProps} />
  );

  describe('when requred props are defined for reply ChatWrapper', () => {
    beforeEach(() => {
      props = {
        user: {
          lastViewedMessage: 1519307926006,
          userId: '5a698c38eeaed821644cfb96',
          username: 'admin'
        },
        messages: [
          '26 марта',
          {
            _id: '5ab8f0559cb87d32f0e5205b',
            content: '17',
            date: '16:06',
            userId: '5a698c38eeaed821644cfb96',
            username: 'admin'
          }
        ],
        selectedRoom: {
          _id: '5ab613f11a9c4a57a48233a1',
          backgroundSrc: 'https://2ch.hk/b/src/173280275/15221806039983.jpg',
          description: '----------------',
          isPrivate: false,
          participants: {
            admin: {
              color: 'a10009'
            }
          },
          roomName: 'test',
          userId: '5a66ea7ba831d74870af4885'
        },
        onCreateMessage: jest.fn(),
        onGetMessages: jest.fn(),
        onGetOneMessage: jest.fn(),
        onSetRoomBackground: jest.fn(),
        onChangeRoomBackground: jest.fn(),
        onResetRoom: jest.fn(),
        onOpenRoom: jest.fn()
      };
    });

    it('renders correctly', () => {
      const wrapper = mount(chatWrapper(props));
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('check logic on unselected room with correct path in history url', () => {
      props.selectedRoom._id = null;
      const wrapper = mount(
        <MemoryRouter initialEntries={['/#/room/5ab613f11a9c4a57a48233a1']}>
          {chatWrapper(props)}
        </MemoryRouter>
      );
      expect(props.onOpenRoom).toHaveBeenCalledTimes(1);
    });

    it('reset room when component is unmounting', () => {
      const wrapper = mount(chatWrapper(props));
      wrapper.unmount();
      expect(props.onResetRoom).toHaveBeenCalledTimes(1);
    });
  });
});
