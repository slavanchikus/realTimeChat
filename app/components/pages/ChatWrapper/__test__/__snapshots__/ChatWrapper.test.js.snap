// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatWrapper when requred props are defined for reply ChatWrapper renders correctly 1`] = `
<ChatWrapper
  messages={
    Array [
      "26 марта",
      Object {
        "_id": "5ab8f0559cb87d32f0e5205b",
        "content": "17",
        "date": "16:06",
        "userId": "5a698c38eeaed821644cfb96",
        "username": "admin",
      },
    ]
  }
  onChangeRoomBackground={[MockFunction]}
  onCreateMessage={[MockFunction]}
  onGetMessages={[MockFunction]}
  onGetOneMessage={[MockFunction]}
  onOpenRoom={[MockFunction]}
  onResetRoom={[MockFunction]}
  onSetRoomBackground={[MockFunction]}
  selectedRoom={
    Object {
      "_id": "5ab613f11a9c4a57a48233a1",
      "backgroundSrc": "https://2ch.hk/b/src/173280275/15221806039983.jpg",
      "description": "----------------",
      "isPrivate": false,
      "participants": Object {
        "admin": Object {
          "color": "a10009",
        },
      },
      "roomName": "test",
      "userId": "5a66ea7ba831d74870af4885",
    }
  }
  user={
    Object {
      "lastViewedMessage": 1519307926006,
      "userId": "5a698c38eeaed821644cfb96",
      "username": "admin",
    }
  }
>
  <div
    className="container"
  >
    <Header
      onSetRoomBackground={[MockFunction]}
      selectedRoom={
        Object {
          "_id": "5ab613f11a9c4a57a48233a1",
          "backgroundSrc": "https://2ch.hk/b/src/173280275/15221806039983.jpg",
          "description": "----------------",
          "isPrivate": false,
          "participants": Object {
            "admin": Object {
              "color": "a10009",
            },
          },
          "roomName": "test",
          "userId": "5a66ea7ba831d74870af4885",
        }
      }
    >
      <div
        className="container"
      >
        <div
          className="left_part"
        >
          <h2>
            test
          </h2>
          <div>
            Онлайн: 0
          </div>
        </div>
        <SideBar
          onSetRoomBackground={[MockFunction]}
          selectedRoom={
            Object {
              "_id": "5ab613f11a9c4a57a48233a1",
              "backgroundSrc": "https://2ch.hk/b/src/173280275/15221806039983.jpg",
              "description": "----------------",
              "isPrivate": false,
              "participants": Object {
                "admin": Object {
                  "color": "a10009",
                },
              },
              "roomName": "test",
              "userId": "5a66ea7ba831d74870af4885",
            }
          }
        >
          <div
            className="container"
          >
            <div
              className="menu_icon"
              onClick={[Function]}
            >
              <div
                className="hamburger"
              />
            </div>
          </div>
        </SideBar>
      </div>
    </Header>
    <MessagesContainer
      messages={
        Array [
          "26 марта",
          Object {
            "_id": "5ab8f0559cb87d32f0e5205b",
            "content": "17",
            "date": "16:06",
            "userId": "5a698c38eeaed821644cfb96",
            "username": "admin",
          },
        ]
      }
      onGetMessages={[MockFunction]}
      selectedRoom={
        Object {
          "_id": "5ab613f11a9c4a57a48233a1",
          "backgroundSrc": "https://2ch.hk/b/src/173280275/15221806039983.jpg",
          "description": "----------------",
          "isPrivate": false,
          "participants": Object {
            "admin": Object {
              "color": "a10009",
            },
          },
          "roomName": "test",
          "userId": "5a66ea7ba831d74870af4885",
        }
      }
      user={
        Object {
          "lastViewedMessage": 1519307926006,
          "userId": "5a698c38eeaed821644cfb96",
          "username": "admin",
        }
      }
    >
      <div
        className="container"
        id="messagescontainer"
        onScroll={[Function]}
        style={
          Object {
            "background": "url(https://2ch.hk/b/src/173280275/15221806039983.jpg) 50% 50% / cover no-repeat rgb(54, 54, 54)",
          }
        }
      >
        <div
          className="day"
          key="26 марта"
        >
          26 марта
        </div>
        <Message
          key="5ab8f0559cb87d32f0e5205b"
          message={
            Object {
              "_id": "5ab8f0559cb87d32f0e5205b",
              "content": "17",
              "date": "16:06",
              "userId": "5a698c38eeaed821644cfb96",
              "username": "admin",
            }
          }
          selectedRoom={
            Object {
              "_id": "5ab613f11a9c4a57a48233a1",
              "backgroundSrc": "https://2ch.hk/b/src/173280275/15221806039983.jpg",
              "description": "----------------",
              "isPrivate": false,
              "participants": Object {
                "admin": Object {
                  "color": "a10009",
                },
              },
              "roomName": "test",
              "userId": "5a66ea7ba831d74870af4885",
            }
          }
          user={
            Object {
              "lastViewedMessage": 1519307926006,
              "userId": "5a698c38eeaed821644cfb96",
              "username": "admin",
            }
          }
        >
          <div
            className="message own_message"
          >
            <span
              className="username"
              style={
                Object {
                  "color": "#a10009",
                }
              }
            >
              admin
            </span>
            <p
              dangerouslySetInnerHTML={
                Object {
                  "__html": "17",
                }
              }
            />
            <span
              className="date"
            >
              16:06
            </span>
          </div>
        </Message>
      </div>
    </MessagesContainer>
    <InputForm
      onCreateMessage={[MockFunction]}
      selectedRoom={
        Object {
          "_id": "5ab613f11a9c4a57a48233a1",
          "backgroundSrc": "https://2ch.hk/b/src/173280275/15221806039983.jpg",
          "description": "----------------",
          "isPrivate": false,
          "participants": Object {
            "admin": Object {
              "color": "a10009",
            },
          },
          "roomName": "test",
          "userId": "5a66ea7ba831d74870af4885",
        }
      }
      user={
        Object {
          "lastViewedMessage": 1519307926006,
          "userId": "5a698c38eeaed821644cfb96",
          "username": "admin",
        }
      }
    >
      <div
        className="container"
      >
        <div
          className="input_wrapper"
        >
          <div
            className="upload"
          >
            <form>
              <input
                accept="image/png, image/jpeg"
                disabled={false}
                name="image"
                onChange={[Function]}
                type="file"
              />
            </form>
          </div>
          <div
            className="input"
            contentEditable={true}
            onBlur={[Function]}
            onFocus={[Function]}
            onInput={[Function]}
            onKeyPress={[Function]}
            onSelect={[Function]}
            spellCheck={false}
          />
          <div
            className="post"
            onClick={[Function]}
          />
        </div>
      </div>
    </InputForm>
  </div>
</ChatWrapper>
`;
